src/components/Tokens/TokenDetails/index.tsx

```diff
@@ -234,13 +234,13 @@ export default function TokenDetails({
 
         <RightPanel onClick={() => isBlockedToken && setOpenTokenSafetyModal(true)}>
           <div style={{ pointerEvents: isBlockedToken ? 'none' : 'auto' }}>
-            <Widget
-              defaultTokens={{
-                [Field.INPUT]: inputToken ?? undefined,
-                default: detailedToken ?? undefined,
+            <Swap
+              chainId={pageChainId}
+              prefilledState={{
+                [Field.INPUT]: { currencyId: inputToken?.wrapped?.address },
+                [Field.OUTPUT]: { currencyId: address === NATIVE_CHAIN_ID ? 'ETH' : address },
```

Comment:
nit: this is unrelated to this PR, but I would consider this a good follow-up task (maybe we can write this down). I believe `Swap` component and `getTokenDetailsURL` should check INSIDE for `address === NATIVE_CHAIN_ID` and use `ETH` or `address` instead of the caller.